CMakeLists.txt

```diff
@@ -34,6 +34,37 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# find all known ROS1 message/service packages
+execute_process(
+    COMMAND "rosmsg" "list"
+    OUTPUT_VARIABLE rosmsg_output
+    ERROR_VARIABLE rosmsg_error
+)
+execute_process(
+    COMMAND "rossrv" "list"
+    OUTPUT_VARIABLE rossrv_output
+    ERROR_VARIABLE rossrv_error
+)
+if(NOT rosmsg_error STREQUAL "" OR NOT rossrv_error STREQUAL "")
+  message(FATAL_ERROR "${rosmsg_error}\n${rossrv_error}\nFailed to call rosmsg/rossrv\nPlease make sure to source your ROS2 workspace BEFORE sourcing your ROS1 workspace")
```

Comment:
Could you clarify why this is necessary in the first place? (In the context of the bridge being built separately). Is it to find the mapping_rules? or is it for another reason that it's necessary to find the ROS2 cmake config files?

The problem I'm facing is that given that the message packages have the same names on both side, the ROS1 python scripts (rosmsg and rossrv) need the ros ones to be the first ones on the sys.path so that when they `import std_msgs` they get the ROS1 module. I'm happy to explore an alternative solution to the rosmsg/rossrv approach if you have a pointer to what would be the best way to handle this.